aleph-client/src/treasury.rs

```diff
@@ -0,0 +1,152 @@
+use crate::{try_send_xt, wait_for_event, AnyConnection, RootConnection, SignedConnection};
+use ac_primitives::ExtrinsicParams;
+use codec::Decode;
+use frame_support::PalletId;
+use primitives::Balance;
+use sp_core::{Pair, H256};
+use sp_runtime::{traits::AccountIdConversion, AccountId32};
+use std::{thread, thread::sleep, time::Duration};
+use substrate_api_client::{compose_extrinsic, ApiResult, GenericAddress, XtStatus};
+
+type AnyResult<T> = anyhow::Result<T>;
+
+/// Returns the account of the treasury.
+pub fn treasury_account() -> AccountId32 {
+    PalletId(*b"a0/trsry").into_account_truncating()
+}
+
+/// Returns how many treasury proposals have ever been created.
+///
+/// Requires a single storage read.
+pub fn proposals_counter<C: AnyConnection>(connection: &C) -> u32 {
+    connection
+        .as_connection()
+        .get_storage_value("Treasury", "ProposalCount", None)
+        .expect("Key `Treasury::ProposalCount` should be present in storage")
+        .unwrap_or(0)
+}
+
+/// Calculates how much balance will be paid out to the treasury after each era.
+///
+/// Every call to this method is potentially expensive - it requires three storage reads.
+pub fn staking_treasury_payout<C: AnyConnection>(connection: &C) -> Balance {
+    let sessions_per_era = connection
+        .as_connection()
+        .get_constant::<u32>("Staking", "SessionsPerEra")
+        .expect("Constant `Staking::SessionsPerEra` should be present");
+    let session_period = connection
+        .as_connection()
+        .get_constant::<u32>("Elections", "SessionPeriod")
+        .expect("Constant `Elections::SessionPeriod` should be present");
+    let millisecs_per_block = 2 * connection
```

Comment:
I know this was in the original code already, but this `2*` is quite sneaky meaning it makes code more coupled - ie technically this line now depends on line https://github.com/Cardinal-Cryptography/aleph-node/blob/main/bin/runtime/src/lib.rs#L525 . On the other hand I don't have an idea of how to make this read from storage so I think this was the only way then

Reply:
I changed it to read directly from `primitives`